app/routes.py

```diff
@@ -1,2 +1,292 @@
+
+############
+def format_goal_dictionary(new_goal):
+    goal = {}
+    goal['id'] = new_goal.goal_id
+    goal['title'] = new_goal.title
+
+    return goal
+############
+
+#Get Goals
+@goals_bp.route("", methods=["GET"])
+def get_goals():
+
+    sort_query = request.args.get("sort")
+    if(sort_query == "asc"):
+        goals = Goal.query.order_by(Goal.title).all()
+    elif(sort_query == "desc"):
+        goals = Goal.query.order_by(Goal.title.desc()).all()
+    else:
+        goals = Goal.query.all()
+    
+    response = []
+
+    for goal in goals:
+        goal_dict = format_goal_dictionary(goal)        
+        response.append(goal_dict)
+
+    return jsonify(response)
+
+#Get a Goal
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    
+    if goal is None:
+        return make_response("Not Found", 404)
+    
+    response = {}
+    response["goal"] = format_goal_dictionary(goal)
+    return jsonify(response)
+
+#Update One Goal
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    
+    if goal is None:
+        return make_response(f"Goal {goal_id} not found", 404)
+    
+    form_data = request.get_json()
+    goal.title = form_data["title"]
+    db.session.commit()
+    
+    response = {}
+    response["goal"] = format_goal_dictionary(goal)
+    return make_response(response, 200)
+
+
+# Delete One Goal
+@goals_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return make_response(f"Goal {goal_id} not found", 404)
+    response = {"details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted' }
+    db.session.delete(goal)
+    db.session.commit()
+    return make_response(response, 200)
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"])
+def post_task_ids_to_goal(goal_id):
+    goal = Goal.query.get(goal_id) #we grabbed the goal for the provided id
+
+    if goal is None: #error checking
+        return make_response("", 404)
+    
+    request_body = request.get_json() #grab form data
+    task_ids = request_body["task_ids"] #storing the list in a cleaner variable reference
+    
+    for task in range(len(task_ids)): #for each index in this list
+        this_task = Task.query.get(task_ids[task]) #get one task at a time
+        this_task.goal_id = goal.goal_id #set its goal id to the id of this goal
+        db.session.commit()
+
+    response = {"id": goal.goal_id, "task_ids": task_ids}
+    return make_response(response, 200)
+
+
+@goals_bp.route("/<goal_id>/tasks", methods=["GET"])
+def get_tasks_for_specific_goals(goal_id): 
+
+    #get the goal cuz we need its details
+    #get tasks that have that goal id 
+    #if no tasks have that goal id, return a specific response 
+
+    goal = Goal.query.get(goal_id) #grab the specific goal
+    if goal is None:
+        return make_response("", 404)
```

Comment:
It would be best to include in the body some indication of why it failed in addition to the response code.